cluster/plan.go

```diff
@@ -558,6 +560,26 @@ func (c *Cluster) BuildKubeletProcess(host *hosts.Host, serviceOptions v3.Kubern
 			fmt.Sprintf("%s:c:/host/run", path.Join(host.PrefixPath, "/run")),
 		}...)
 	} else {
+		parsedVersion, err := getClusterVersion(c.Version)
+		if err != nil {
+			logrus.Debugf("Error while parsing cluster version: %s", err)
+		}
+		parsedJanK8s123Version, err := getClusterVersion(JanK8s123Version)
+		if err != nil {
+			logrus.Debugf("Error while parsing cluster version: %s", err)
+		}
+		parsedAprilK8s124Version, err := getClusterVersion(AprilK8s124Version)
+		if err != nil {
+			logrus.Debugf("Error while parsing cluster version: %s", err)
+		}
+		parsedAprilK8s125Version, err := getClusterVersion(AprilK8s125Version)
+		if err != nil {
+			logrus.Debugf("Error while parsing cluster version: %s", err)
+		}
+		if parsedVersion.GE(parsedJanK8s123Version) && parsedVersion.GE(parsedAprilK8s124Version) && parsedVersion.GE(parsedAprilK8s125Version) {
```

Comment:
I don't believe this logic is correct. For example, any `1.24.x` version will result in `false` returned by `parsedVersion.GE(parsedAprilK8s125Version)` and since all parts are AND-ed will result in `false` overall.
What we need here is to check ranges similar to how we do in KDM for applying different templates, e.g. checking against ranges like `>=1.24.13-rancher1-1 <1.25.0-rancher1-1` with all checks OR-ed.

Reply:
@doflamingo721 , this is not sufficient to just "OR" the existing conditions as after this change any `1.24.x` version will return `true` because it's greater than `v1.23.16-rancher1-1`.
The logic we need is "any 1.23.x version that is greater or equal than v1.23.16-rancher1-1 OR any 1.24.x version that is greater or equal than v1.24.13-rancher1 OR any 1.25.x version that is greater or equal than v1.25.9-rancher1-1" - there are likely multiple ways to implement it with one being switching to matching **ranges** like `>=1.24.13-rancher1-1 <1.25.0-rancher1-1` as suggested in the original comment.

Reply:
Okay. I understood it now. Will update accordingly. Thanks for the explanation.